src/ray/raylet/scheduling/fair_scheduling_queue_test.cc

```diff
@@ -0,0 +1,142 @@
+#include "ray/raylet/scheduling/fair_scheduling_queue.h"
+#include "ray/raylet/scheduling/test_util.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace ray {
+
+namespace raylet {
+
+using ::testing::_;
+
+class FairSchedulingQueueTest : public ::testing::Test {
+ public:
+  FairSchedulingQueue queue_;
+
+  Work TaskToWork(Task &task) { return std::make_tuple(task, nullptr, nullptr); }
+
+  void AssertNoLeaks() {
+    ASSERT_EQ(queue_.active_tasks_.size(), 0);
+    ASSERT_EQ(queue_.work_queue_.size(), 0);
+  }
+};
```

Comment:
Can you have a separate test to test `work_queue`?  You can verify

- if exposed methods are working
- if priority orders are calculated correctly.